lib/donation_system/paypal_wrapper/creator_fields_generator.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+require 'money'
+
+require_relative '../result'
+require_relative '../utils'
+
+module DonationSystem
+  module PaypalWrapper
+    class CreatorFieldsGenerator
+      INTENT = 'sale'
+      PAYPAL_METHOD = 'paypal'
+      NUMBER_PREFIX = 'P'
+
+      def self.execute(data)
+        new(data).execute
+      end
+
+      def initialize(data)
+        @data = data
+      end
+
+      def execute
+        {
+          intent: INTENT,
+          payer: { payment_method: PAYPAL_METHOD },
+          redirect_urls: redirect_urls,
+          transactions: [{
+            amount: { total: amount, currency: currency },
+            description: utils.generate_number(NUMBER_PREFIX)
+          }]
+        }
+      end
+
+      private
+
+      attr_reader :data
+
+      def amount
+        I18n.enforce_available_locales = false
+        Money.from_amount(BigDecimal(data.amount).abs, data.currency).to_s
+      end
+
+      def currency
+        data.currency.upcase
```

Comment:
This slightly different from how currency is retrieved in `CreatorDataValidator`. Should it not get the currency symbol via `amount`? (e.g. via `Money`)

Reply:
This is fixed in a future PR that includes refactorings